lib/connection/pool.js

```diff
@@ -907,6 +925,25 @@ Pool.prototype.destroy = function(force) {
   checkStatus();
 }
 
+// Facilitate compressing a message using an agreed compressor
+var compress = function(self, dataToBeCompressed, callback) {
```

Comment:
it's starting to feel like we might need a separate file in `mongodb-core/lib/wireprotocol` called `compression.js` which encapsulates all this code that relates to compression. Thoughts? (`compress`, `decompress`, `compressorIds`, `uncompressibleCommands`, etc)